kindelia_core/src/events.rs

```diff
@@ -690,149 +689,6 @@ macro_rules! heartbeat {
-//       if let Err(err) = ws_tx.send(event.clone()) {
-//         eprintln!("Could not send event to websocket: {}", err);
-//       };
-//     }
-//   };
-//   let websocket_send = Box::new(websocket_send);
-
-//   let file = |event: NodeEvent| {
-//     let txt = format!("{},\n", serde_json::to_string(&event.clone()).unwrap());
-//     // saves it in a file
-//     std::fs::
-//     file.write_all(txt.as_bytes()).unwrap();
-//   };
-//   let file = Box::new(file);
-
-//   vec![print, websocket_send, file]
-// }
-
-// =======================================================
-// Websocket server
-
-#[derive(serde::Deserialize)]
-pub struct QueryParams {
-  pub tags: Option<String>,
-}
-
-pub struct Query {
-  pub tags: Vec<String>,
-}
-
-pub fn ws_loop(port: u16, ws_tx: broadcast::Sender<NodeEventType>) {
-  let runtime = tokio::runtime::Runtime::new().unwrap();
-  runtime.block_on(async move {
-    ws_server(port, ws_tx).await;
-  });
-}
-
-async fn ws_server(port: u16, ws_tx: broadcast::Sender<NodeEventType>) {
-  let ws_route = warp::ws()
-    .and(with_rx(ws_tx.clone()))
-    .and(warp::query::<QueryParams>().map(parse_query))
-    .and_then(ws_handler);
-  warp::serve(ws_route).run(([127, 0, 0, 1], port)).await;
-}
-
-fn parse_query(query: QueryParams) -> Query {
-  let tags = match query.tags {
-    Some(tags) => tags.split(',').map(str::to_string).collect(),
-    None => vec![],
-  };
-  Query { tags }
-}
-
-fn with_rx(
-  ws_tx: broadcast::Sender<NodeEventType>,
-) -> impl Filter<Extract = (broadcast::Sender<NodeEventType>,), Error = Infallible>
-     + Clone {
-  warp::any().map(move || ws_tx.clone())
-}
-
-pub async fn ws_handler(
-  ws: warp::ws::Ws,
-  ws_tx: broadcast::Sender<NodeEventType>,
-  query: Query,
-) -> Result<impl Reply, Rejection> {
-  Ok(ws.on_upgrade(move |socket| client_connection(socket, ws_tx, query.tags)))
-}
-
-pub async fn client_connection(
-  ws: WebSocket,
-  ws_tx: broadcast::Sender<NodeEventType>,
-  tags: Vec<String>,
-) {
-  let (mut client_ws_sender, _) = ws.split();
-  let mut ws_rx = ws_tx.subscribe();
-  let mut count = 0;
-
-  while let Ok(event) = ws_rx.recv().await {
-    let tags: Result<Vec<NodeEventDiscriminant>, String> =
-      tags.iter().map(|tag| NodeEventDiscriminant::from_str(tag)).collect();
-
-    if let Ok(tags) = tags {
-      if tags.is_empty() || tags.contains(&(&event).into()) {
-        let json_stringfied = serde_json::to_string(&event).unwrap();
-        if let Err(err) =
-          client_ws_sender.send(Message::text(json_stringfied)).await
-        {
-          eprintln!("Could not send message through websocket: {}", err);
-          count += 1;
-        } else {
-          count = 0;
-        };
-        // After 10 consecutive fails we close the connection
-        if count == 10 {
-          break;
-        };
-      }
-    } else {
-      break;
-    }
-  }
-
-  eprintln!("Disconnected");
-}
-
-//
-
-// TODO
 pub fn spawn_event_handlers<A: ProtoAddr + 'static>(
```

Comment:
This thread spawning code will be on `kindelia` or `kindelia_ws`. It will be glued with the code Node by `kindelia`.